engine/src/main/java/io/zeebe/engine/processing/bpmn/behavior/BpmnEventSubscriptionBehavior.java

```diff
@@ -59,24 +61,26 @@
   private final MutableElementInstanceState elementInstanceState;
   private final CatchEventBehavior catchEventBehavior;
 
-  private final TypedStreamWriter streamWriter;
+  private final StateWriter stateWriter;
   private final SideEffects sideEffects;
 
   private final KeyGenerator keyGenerator;
   private final ProcessState processState;
   private final MutableVariableState variablesState;
+  private final TypedCommandWriter commandWriter;
 
   public BpmnEventSubscriptionBehavior(
       final BpmnStateBehavior stateBehavior,
       final BpmnStateTransitionBehavior stateTransitionBehavior,
       final CatchEventBehavior catchEventBehavior,
-      final TypedStreamWriter streamWriter,
+      final Writers writers,
```

Comment:
Just a nitpick, but the class is really dependent only on the state and command writer. I get that it's more convenient to pass the whole thing, but if we want to use constructors to indicate more clearly the dependencies, it probably makes sense to pass the direct ones.

If we have too many arguments, then it's probably a good flag that our class is doing too much :sweat_smile:

Reply:
Yeah you're right. I will remove the writers dependency.